### ISSUE TITLE
Real-valued Fourier transform

### ISSUE THREAD

**Issue by crisluengo** [DIPlib/diplib#27]
It would be nice to have, for the internal Fourier transform code, specific paths for real input or output, which could mean a 2x increase in performance. When compiled with *FFTW*, such paths already exist.

**Comment by crisluengo** [comment]
This is (partially) implemented in ef536b709cc8be36baa289f7721ba38d04f7b21e

What this does is apply a normal transform to each line along the first transform dimension, then for the other transform dimensions it only processes the non-redundant lines. At the end it copies (with complex conjugation) the computed values to the other half of the image.

The 2x improvement in performance is not really there. For the 2D image sizes I tried, the performance is about equal. In 3D the difference becomes more significant.

Note that the real-to-complex transform that exists in FFTW doesn't seem any faster than the complex-to-complex transform for "normal" image sizes. In fact, it seems a little slower. In fact, the first implementation of `dip::FourierTransform` using FFTW, which simply prepared the image and called FFTW's nD R2C FFT or C2C FFT as appropriate, showed that the R2C FFT was significantly slower than than the C2C FFT for normal image sizes. With the new implementation (which always uses the 1D C2C FFT in FFTW), we're probably close to the optimal solution. (Note also that running the 1D C2C FFT from FFTW in parallel is more efficient than running the nD C2C FFT in FFTW using FFTW's multithreading. FFTW doesn't seem to get much benefit from multithreading, I think they might multithread each of the 1D FFTs, rather than running the many 1D FFTs in parallel.)
